software/src/8.0/src/kernel/VkUUID.cpp

```diff
@@ -58,16 +31,43 @@ typedef caddr_t ifc_buf_t;
  *******************************************
  *******************************************/
 
+typedef V::uuid_t::u32	u32;
+typedef V::uuid_t::u16	u16;
+typedef V::uuid_t::u8	u8;
+
 /***********************************
  *****  DCE Compliant Systems  *****
  ***********************************/
 
-#if defined(_AIX) || defined(__hpux)
-#define DCE_UUIDGEN
+#if defined(_WIN32)
+#define USING_WINDOWS_DCE
+#include <rpcdce.h>
+#pragma comment(lib, "rpcrt4.lib")
 
-typedef uuid_t::u32	u32;
-typedef uuid_t::u16	u16;
-typedef uuid_t::u8	u8;
+namespace {
+    uuid_t* UCast (V::uuid_t const &rUUID) {
+	return (uuid_t*)(&rUUID);
+    };
+    uuid_t* UCast (V::uuid_t &rUUID) {
+	return (uuid_t*)(&rUUID);
+    };
+}
+
+#elif defined(__APPLE__) || defined(__linux__) || defined(sun)
+#define USING_LIBUUID_DCE
+#include <uuid/uuid.h>
```

Comment:
That ought to be a reasonable requirement.  libuuid is part of the Linux runtime and should provide higher quality UUID generation than what we've got.  I'm not sure why RedHat left out the part (libuuid-devel) that contains the include files.  They seem to be present elsewhere.

Don't know how much I'll be able to look, but what object files/libraries are being implicated by the linker?  Is this a clean build?